exercises/robot-simulator/src/robot_simulator.h

```diff
@@ -0,0 +1,40 @@
+#ifndef _ROBOT_SIMULATOR_H
+#define _ROBOT_SIMULATOR_H
+
+typedef enum {
+   HeadingNorth = 0,
+   HeadingEast,
+   HeadingSouth,
+   HeadingWest,
+   HeadingMax
+} Bearing_t;
+
+enum {
+   Default_Bearing = HeadingNorth,
+   Default_X_Coordinate = 0,
+   Default_Y_Coordinate = 0,
+};
+
+enum {
+   Command_Left = 'L',
+   Command_Right = 'R',
+   Command_Advance = 'A'
+};
+
+typedef struct RobotCoordinates {
+   int x_position;
+   int y_position;
+} Robott_Coordinates_t;
+
+typedef struct RobotGridStatus {
+   Bearing_t bearing;
+   Robott_Coordinates_t grid;
+} RobotGridStatus_t;
+
+RobotGridStatus_t robot_init(int args, ...);
```

Comment:
This is fairly opaque. Why not have two `init` functions or have some default value that can be passed in?